plugins/inputs/huebridge/bridge.go

```diff
@@ -0,0 +1,323 @@
+package huebridge
+
+import (
+	"fmt"
+	"math"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"slices"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/tdrn-org/go-hue"
+)
+
+type bridge struct {
+	url             *url.URL
+	roomAssignments map[string]string
+	rcc             *RemoteClientConfig
+	tcc             *tls.ClientConfig
+	timeout         config.Duration
+	log             telegraf.Logger
+	resolvedClient  hue.BridgeClient
+}
+
+func newBridge(rawUrl string, roomAssignments map[string]string, rcc *RemoteClientConfig, tcc *tls.ClientConfig, timeout config.Duration, log telegraf.Logger) (*bridge, error) {
+	parsedUrl, err := url.Parse(rawUrl)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse bridge URL %s: %w", rawUrl, err)
+	}
+	validSchemes := []string{"address", "cloud", "mdns", "remote"}
+	if slices.Index(validSchemes, parsedUrl.Scheme) < 0 {
+		return nil, fmt.Errorf("unrecognized scheme %s in URL %s", parsedUrl.Scheme, parsedUrl)
+	}
+	// All schemes require a password in the URL
```

Comment:
I think this is easier to read
```suggestion
	switch parsedUrl.Scheme {
	case "address", "cloud", "mdns", "remote":
		// Do nothing, those are valid
	default:
		return nil, fmt.Errorf("unrecognized scheme %s in URL %s", parsedUrl.Scheme, parsedUrl)
	}

	// All schemes require a password in the URL
```

Reply:
Adapted via 62d8804ee8396401842e4c93008d15d441651751